### ISSUE TITLE
Screencast not working [lineage-16.0 | lineage-17.1]

### ISSUE THREAD

**Issue by AlbertoZandara** [BYZYB/android_device_xiaomi_libra#12]
If you try to connect the phone to a wireless chromcast device, using the buit-in function "Screencast", the OS crashes and the smartphone reboots.

**Comment by BYZYB** [comment]
### Related logcat
```
09-21 18:20:09.203   654   704 I WifiDisplayController: Connected to Wifi display: DESKTOP-XXXXXXX
09-21 18:20:09.203   654   704 I WifiDisplayController: Stopping Wifi display scan.
09-21 18:20:09.207   654   704 I WifiDisplayController: Listening for RTSP connection on 192.168.137.247:7236 from Wifi display: DESKTOP-XXXXXXX
09-21 18:20:09.208   494  2397 E MediaPlayerService: listenForRemoteDisplay is no longer supported!
09-21 18:20:09.209   654   704 E RemoteDisplay: Media player service rejected request to listen for remote display '192.168.137.247:7236'.
--------- beginning of crash
09-21 18:20:09.209   654   704 E AndroidRuntime: *** FATAL EXCEPTION IN SYSTEM PROCESS: android.display
09-21 18:20:09.209   654   704 E AndroidRuntime: java.lang.IllegalStateException: Could not start listening for remote display connection on "192.168.137.247:7236"
09-21 18:20:09.209   654   704 E AndroidRuntime: 	at android.media.RemoteDisplay.startListening(RemoteDisplay.java:123)
09-21 18:20:09.209   654   704 E AndroidRuntime: 	at android.media.RemoteDisplay.listen(RemoteDisplay.java:85)
09-21 18:20:09.209   654   704 E AndroidRuntime: 	at com.android.server.display.WifiDisplayController.updateConnection(WifiDisplayController.java:819)
09-21 18:20:09.209   654   704 E AndroidRuntime: 	at com.android.server.display.WifiDisplayController.access$1200(WifiDisplayController.java:70)
09-21 18:20:09.209   654   704 E AndroidRuntime: 	at com.android.server.display.WifiDisplayController$15.onGroupInfoAvailable(WifiDisplayController.java:916)
09-21 18:20:09.209   654   704 E AndroidRuntime: 	at android.net.wifi.p2p.WifiP2pManager$Channel$P2pHandler.handleMessage(WifiP2pManager.java:985)
09-21 18:20:09.209   654   704 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
09-21 18:20:09.209   654   704 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:214)
09-21 18:20:09.209   654   704 E AndroidRuntime: 	at android.os.HandlerThread.run(HandlerThread.java:67)
09-21 18:20:09.209   654   704 E AndroidRuntime: 	at com.android.server.ServiceThread.run(ServiceThread.java:44)
09-21 18:20:09.210   654   704 I Process : Sending signal. PID: 654 SIG: 9
09-21 18:20:09.430   347   347 I ServiceManager: all services died (NOTE: skipped duplicate logs about "service died")
09-21 18:20:09.543  2177  2177 D AndroidRuntime: Shutting down VM
09-21 18:20:09.544  2177  2177 E AndroidRuntime: FATAL EXCEPTION: main
09-21 18:20:09.544  2177  2177 E AndroidRuntime: Process: com.android.systemui, PID: 2177
09-21 18:20:09.544  2177  2177 E AndroidRuntime: DeadSystemException: The system died; earlier logs will point to the root cause
```

### Related dmesg
Not including duplicated logs.
```
[ 1907.190606] ------------[ cut here ]------------
[ 1907.190626] WARNING: at ../../../../../../kernel/xiaomi/libra/kernel/power/qos.c:605 0xffffffc0007df48c()
[ 1907.190628] pm_qos_update_request() called for unknown object
[ 1907.190635] CPU: 1 PID: 13470 Comm: cameraserver Tainted: G        W    3.10.108-ga3fb3318b21 #2
[ 1907.190637] Hardware name: Qualcomm Technologies, Inc. MSM8992 MTP (DT)
[ 1907.190640] Call trace:
[ 1907.190642] [<ffffffc00008f4f0>] 0xffffffc00008f4f0
[ 1907.190646] ---[ end trace 48e137f4cad9f383 ]---
```

**Comment by BYZYB** [comment]
> If you try to connect the phone to a wireless chromcast device, using the buit-in function "Screencast", the OS crashes and the smartphone reboots.

Native ScreenCast/MiraCast support has been dropped in Pie (as well as 10/11). There is some custom WFD solution for msm8996, but none of them is suitable for msm8992/8994. As a result, WFD support should be disabled in lineage-16.0/17.1/18.0 roms: e89d8d9f7a10a05b64ecee1d93507f5e8aff6466.
This issue cannot be fixed currently, sorry for the inconvenience.

**Comment by BYZYB** [comment]
WFD has been fixed in the latest build of lineage-18.1 ([download](https://github.com/BYZYB/android_device_xiaomi_libra/releases/tag/v2021.10.17)).
